app/routes.py

```diff
@@ -1,2 +1,231 @@
+        if video == None:
+            return make_response("",404)        
+        else:
+            db.session.delete(video)
+            db.session.commit()
+            return make_response ({"id": video.video_id}), 200
+
+# This route will check out a video to a customer, and update the data in the database 
+@rental_bp.route("/check-out", methods=["POST"])
+def checkout_video():
+    request_body = request.get_json()
+
+    # Required Request Body Parameters
+    customer = Customer.query.get(request_body["customer_id"])
+    video = Video.query.get(request_body["video_id"])
+    # check if customer id and video id are int
+    if type(customer) != int or type(video) != int:
+        return jsonify({"details":"Invalid data"}, 404)    
+    # return back detailed errors/ status 404: Not Found if the customer or video does not exist
+    if customer == None or video == None:
+        return jsonify({"details":"Invalid data"}, 404)
+    # return back detailed errors/status 400: Bad Request if the video does not have any available inventory before check out    
+    if video.available_inventory==0:
+        return jsonify({"details":"Video out of stock"}, 400)    
+    
+    else:
+        new_rental= Rental(customer_id=request_body["customer_id"],
+                    video_id=request_body["video_id"],
+                    due_date=datetime.now() + timedelta(days=7))
+        # i dk how i can return the customer and video inventory ????
+        ## increase the customer's videos_checked_out_count by one
+        customer.videos_checked_out_count += 1
+        ## decrease the video's available_inventory by one
+        video.available_inventory -= 1
+        
+        db.session.add(new_rental)
+        db.session.add(customer)
+        db.session.add(video)
+        db.session.commit()
+
+        return make_response(new_rental.to_json_rental(), 200)     
+
+# This route checks in a video to a customer, and updates the data in the database
+@rental_bp.route("/check-in", methods=["POST"])
+def customer_check_in():
+    # look up an explanation on what this is doing ????
+    request_body = request.get_json()
+    video_id = checkin_list.get(id)
+
+    # Required Request Body Parameters
+    customer = Customer.query.get(request_body["customer_id"])
+    video = Video.query.get(request_body["video_id"])
+
+    for rental in customer.video:
+        if rental.video_id == video_id:
+        ## decrease the customer's videos_checked_out_count by one
+            customer.videos_checked_out_count -= 1
+            ## increase the video's available_inventory by one
+            video.available_inventory += 1
+            db.session.commit()
+            return make_response(to_json_rental(), 200)  
+
+
+    else:
+        # return back detailed errors/ status 404: Not Found if the customer or video does not exist
+        if customer == None or video == None:
+            return jsonify({"details":"Invalid data"}, 404)
```

Comment:
We should move this 404 checking to immediately after we query to look for these in the database.